### PR TITLE
fix(limit orders): added proper ETH management for fees

### PR DESCRIPTION
**Pull request by JChiaramonte7**
### Changes:
- Added a way to withdraw ETH from the contract
- Made order placement payable (so people can top up their margin accounts for keeper execution)
- Switched everything to custom errors

### CODE REVIEW

File: contracts/MarginBase.sol [hunk 1]
```diff
@@ -133,6 +133,15 @@ contract MarginBase is MinimalProxyable, OpsReady {
     /// @param required: amount of margin asset required
     error InsufficientFreeMargin(uint256 available, uint256 required);
 
+    /// @notice size cannot be zero when placing a limit/stop order
+    error SizeCannotBeZero();
```
Comment (JaredBorders) [inline contracts/MarginBase.sol:137]: Might want to consider merging this SizeCannotBeZero with the AmountCannotBeZero -> ValueCannotBeZero. Then add it to the modifier I created that was merged into main

File: contracts/MarginBase.sol [hunk 2]
```diff
@@ -558,7 +584,10 @@ contract MarginBase is MinimalProxyable, OpsReady {
         int256 _marginDelta,
         int256 _sizeDelta,
         uint256 _limitPrice
-    ) external onlyOwner returns (uint256) {
+    ) external payable onlyOwner returns (uint256) {
```
Comment (JaredBorders) [inline contracts/MarginBase.sol:587]: Could use modifier here that I proposed above

**Review by JaredBorders — COMMENTED** [review]
Looks good! Some merge conflicts with main because of changes I made but there shouldn't be but a few

File: contracts/MarginBase.sol (hunk 1 above)
Comment (JChiaramonte7) [reply contracts/MarginBase.sol]: Ended up adding an additional param to `ValueCannotBeZero` that way if there's multiple parameters to a function it's clear which one the error is referring to

**Review by JaredBorders — APPROVED** [review]

**Pull request merged**